src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/food/HeavyCream.java

```diff
@@ -28,18 +29,16 @@ public HeavyCream(Category category, SlimefunItemStack item, RecipeType recipeTy
     @Nonnull
     @Override
     public ItemUseHandler getItemHandler() {
-        return this::use;
-    }
-
-    public void use(PlayerRightClickEvent e) {
-        if (e.getClickedBlock().isPresent()) {
-            if (!e.getClickedBlock().get().getType().isInteractable()) {
+        return e -> {
+            Optional<Block> block = e.getClickedBlock();
+            if (block.isPresent()) {
+                if (!block.get().getType().isInteractable()) {
+                    e.cancel();
+                }
+            } else {
                 e.cancel();
             }
```

Comment:
> I don't think i can use && here due to that else statement, correct me if i'm wrong though

Yes buuuuuut what cou could do is the following.
```suggestion
        return e -> {
            Optional<Block> block = e.getClickedBlock();

            if (!block.isPresent() || !block.get().getType().isInteractable()) {
                e.cancel();
            }
```